modules/GKB/CommonUtils.pm

```diff
@@ -112,204 +114,259 @@ sub restore_database {
+    # try the old logic.
+    if ($instance->is_a('Event')) {
+        return $instance->evidenceType->[0] && $instance->evidenceType->[0]->displayName =~ /electronic/i;
+    } elsif ($instance->is_a('PhysicalEntity')) {
+        return !in_gk_central($instance) && has_human_source_instance($instance);
+    } elsif ($instance->is_a('CatalystActivity')) {
+        return any { is_electronically_inferred($_)} @{$instance->reverse_attribute_value('catalystActivity')};
+    } elsif ($instance->is_a('Regulation')) {
+        # For backwards compatability when regulations contained the reaction like event (RLE) they regulated
+        # As of Reactome version 65, RLEs have contained their regulation instances in the 'regulatedBy' attribute
+        my @regulated_events = $instance->regulatedEntity->[0] ? 
+            @{$instance->regulatedEntity} : @{$instance->reverse_attribute_value('regulatedBy')};
+
+        return any { is_electronically_inferred($_)} @regulated_events;
+    }
+}
+
+sub in_gk_central {
+    my $instance = shift;
+
+    state $gk_central_physical_entity_lookup = fetch_gk_central_physical_entity_lookup();
+
+    my $gk_central_instance = $gk_central_physical_entity_lookup->{$instance->db_id};
+
+    return $gk_central_instance &&
+        $instance->class eq $gk_central_instance->class &&
+        $instance->displayName eq $gk_central_instance->displayName &&
+        have_same_species($instance, $gk_central_instance);
+}
+
+sub have_same_species {
+    my $first_instance = shift;
+    my $second_instance = shift;
+
+    my @first_instance_species = map {$_->displayName} @{$first_instance->species};
+    my @second_instance_species = map {$_->displayName} @{$second_instance->species};
+
+    if (scalar @first_instance_species != scalar @second_instance_species) {
+        return 0;
+    }
+
+    foreach my $first_instance_species (@first_instance_species) {
+        # The same species list do not match if the first instance's currently
+        # considered species is not matched to any species in the second
+        # instance
+        return 0 if (!(any {$_ eq $first_instance_species} @second_instance_species));
+    }
+
+    # The species lists must match if they have the same size and all species in
+    # the first list are found in the second list
+    return 1;
+}
+
+sub has_human_source_instance {
+    my $instance = shift;
+
+    # If manually inferred physical entities ever start using the inferredFrom slot, this logic
+    # can be augmented to check the event(s) to which the physical entity is/are attached
+    # to see if the event(s) are electronically inferred
+    return $instance->inferredFrom->[0] &&
+        $instance->inferredFrom->[0]->species->[0] &&
+        $instance->inferredFrom->[0]->species->[0]->displayName =~ /^Homo sapiens$/i;
+}
+
+sub fetch_gk_central_physical_entity_lookup {
+    my $dba = get_dba('gk_central', 'reactomecurator.oicr.on.ca');
```

Comment:
Are the user/pass credentials for gk_central hard-coded somewhere?

Reply:
This actually is addressed in the release-pipeline pull request that's open.  

It uses values in the Config.pm module which are taken from the Secrets.pm module that is not in  the repo.  I originally opened the pull request just to see how the changes look compared to the master branch, but I think it's ready for review.  

Could you have a look at that one and I'll merge it first?  I can then reopen this one so only relevant changes are there.